content/200-concepts/200-database-connectors/08-cockroachdb.mdx

```diff
@@ -0,0 +1,71 @@
+preview: true
+---
+
+<TopBlock>
+
+The CockroachDB data source connector connects Prisma to a [CockroachDB](https://www.cockroachlabs.com/) database server.
+
+</TopBlock>
+
+## Example
+
+To connect to a CockroachDB database server, you need to configure a [`datasource`](/concepts/components/prisma-schema/data-sources) block in your [Prisma schema file](/concepts/components/prisma-schema):
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "cockroachdb"
+  url      = env("DATABASE_URL")
+}
+```
+
+The fields passed to the `datasource` block are:
+
+- `provider`: Specifies the `cockroachdb` data source connector.
+- `url`: Specifies the [connection URL](#connection-details) for the CockroachDB database server. In this case, an [environment variable is used](/concepts/components/prisma-schema#accessing-environment-variables-from-the-schema) to provide the connection URL.
+
+Because CockroachDB support is currently a preview feature, you also need to include it in the list of preview features in the [`generator`]() block:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["cockroachdb"]
+}
+```
+
+## Connection details
+
+CockroachDB uses the PostgreSQL format for its connection URL. See the [PostgreSQL connector documentation](/concepts/database-connectors/postgresql#connection-details) for details of this format, and the optional arguments it takes.
+
+## Type mapping between CockroachDB and the Prisma schema
+
+The CockroachDB connector maps the [scalar types](/concepts/components/prisma-schema/data-model#scalar-fields) from the Prisma [data model](/concepts/components/prisma-schema/data-model) to native column types. These native types are mostly the same as for PostgreSQL — see the [PostgreSQL connector documentation](/concepts/database-connectors/postgresql#type-mapping-between-postgresql-to-prisma-schema) for details. However, there are some limitations, listed below.
+
+> Alternatively, see the [Prisma schema
+> reference](/reference/api-reference/prisma-schema-reference#model-field-scalar-types)
+> for type mappings organized by Prisma type.
+
+### Native type mapping limitations
+
+When introspecting a CockroachDB database, the database types are mapped to Prisma in the same way as for PostgreSQL, with the following limitations:
+
+| CockroachDB (Type \| Aliases) | Prisma    | Supported | Native database type attribute | Notes                                                    |
+| ----------------------------- | --------- | :-------: | :----------------------------- | :------------------------------------------------------- |
+| `money`                       | `Decimal` |  Not yet  | `@db.Money`                    | Supported in PostgreSQL but not currently in CockroachDB |
+| `xml`                         | `String`  |  Not yet  | `@db.Xml`                      | Supported in PostgreSQL but not currently in CockroachDB |
```

Comment:
The notes here could use the CockroachDB issue as a link target of "not currently in CockroachDB" for context.

Reply:
Done